keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -312,13 +318,27 @@ def __init__(
             self.vocabulary = [
                 line.rstrip() for line in tf.io.gfile.GFile(vocabulary)
             ]
+            if vocabulary_size is None:
+                vocabulary_size = len(self.vocabulary)
+            if len(self.vocabulary) < vocabulary_size:
+                logging.warn(
+                    "Setting vocab size to a larger value than the input vocabulary file.\
+￼                        Some token ids will never be output from the tokenizer."
+                )
+                vocabulary_size = len(self.vocabulary)
+            else:
+                self.vocabulary = self.vocabulary[:vocabulary_size]
```

Comment:
If we skip the warning, we can actually reduce this whole block to

```
if vocabulary_size is not None:
    vocabulary_size = min(vocabulary_size, len(self.vocabulary))
    self.vocabulary = self.vocabulary[:vocabulary_size]
```

Will have to wait for @mattdangerw's opinion on this